File: hw/robalino/u3/Hw 37 Strategy/Strategy/src/main/java/ec/edu/espe/StrategySorting/model/QuickSort.java
package ec.edu.espe.StrategySorting.model;

/**
 *
 * @author Benjamin R
 */
public class QuickSort implements SortingStrategy{

    @Override
    public int[] sort(int[] data) {
        int n = data.length;
        for (int i = 1; i < n; i++) {
            int key = data[i];  // Elemento actual a insertar
            int j = i - 1;

            // Mover los elementos mayores que "key" una posición adelante
            while (j >= 0 && data[j] > key) {
                data[j + 1] = data[j];
                j--;
            }
            
            // Insertar "key" en su posición correcta
            data[j + 1] = key;
        }
        return data;
    }
}

File: EXAM/gongora/u3/GongoraStrategy/pooCrudjs/public/controller/QuickSort.js
import  SortingStrategy from "./SortingStrategy.js"

export default class QuickSort extends SortingStrategy { 
    sort(data) {
        let n = data.length;
        for (let i = 1; i < n; i++) {
            let key = data[i];
            let j = i - 1;

            
            while (j >= 0 && data[j] > key) {
                data[j + 1] = data[j];
                j--;
            }

            
            data[j + 1] = key;
        }
        return data;
    }
}


